### PR TITLE
Move JSON parse / stringify operations into runtime

### PR DESCRIPTION
**Pull request by zherczeg**
This patch is mostly code moving. Windows support is speculative. This patch should improve the code quality of JSON module parsing ( #1117 )

### CODE REVIEW

File: src/builtins/BuiltinJSON.cpp [hunk 1]
```diff
@@ -18,698 +18,23 @@
...
-#define RAPIDJSON_ERROR_CHARTYPE char
-#include <rapidjson/document.h>
-#include <rapidjson/filereadstream.h>
-#include <rapidjson/memorystream.h>
-#include <rapidjson/internal/dtoa.h>
-#include <rapidjson/stringbuffer.h>
-#include <rapidjson/writer.h>
-#include <rapidjson/error/en.h>
+#include "runtime/VMInstance.h"
+#include "runtime/Value.h"
```
Comment (clover2123) [inline src/builtins/BuiltinJSON.cpp:26]: This `Value.h` can be removed because `JSON.h` already includes this header file.

**Comment by clover2123** [comment]
BTW why did you move these two JSON functions?
Just for readability?

**Comment by zherczeg** [comment]
For #1117 . If that patch is not needed, this is unnecessary.

**Review by ksh8281 — APPROVED** [review]

**Review by clover2123 — APPROVED** [review]
LGTM

**Pull request merged**